cot/src/config.rs

```diff
@@ -748,6 +759,205 @@ impl LiveReloadMiddlewareConfigBuilder {
+/// let cache_config = SessionStoreTypeConfig::Cache {
+///     uri: CacheUrl::from("redis://localhost:6379"),
+/// };
+/// ```
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", tag = "type")]
+pub enum SessionStoreTypeConfig {
+    /// In-memory session storage.
+    ///
+    /// This uses a simple in-memory store that does not persist sessions across
+    /// application restarts. This is the default, and is suitable for
+    /// development or testing environments.
+    #[default]
+    Memory,
+
+    /// Database-backed session storage.
+    ///
+    /// This stores session data in the configured database. This requires the
+    /// "db" feature to be enabled.
+    #[cfg(feature = "db")]
+    Database,
+
+    /// File-based session storage.
+    ///
+    /// This stores session data in files on the local filesystem. The path to
+    /// the directory where the session files will be stored must be specified.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::path::PathBuf;
+    ///
+    /// use cot::config::SessionStoreTypeConfig;
+    ///
+    /// let config = SessionStoreTypeConfig::File {
+    ///     path: PathBuf::from("/tmp/sessions"),
+    /// };
+    /// ```
+    #[cfg(feature = "json")]
+    File {
+        /// The path to the directory where session files will be stored.
+        path: PathBuf,
+    },
+
+    /// Cache-based session storage.
+    ///
+    /// This stores session data in a cache service like Redis. The URI to the
+    /// cache service must be specified.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::config::{CacheUrl, SessionStoreTypeConfig};
+    ///
+    /// let config = SessionStoreTypeConfig::Cache {
+    ///     uri: CacheUrl::from("redis://localhost:6379"),
+    /// };
+    /// ```
+    #[cfg(feature = "cache")]
+    Cache {
+        /// The URI to the cache service.
+        uri: CacheUrl,
+    },
+}
+
+impl ToSessionStore for SessionStoreTypeConfig {
+    #[expect(unused_variables)]
+    fn to_session_store(
+        self,
+        context: &ProjectContext<WithDatabase>,
+    ) -> Result<Box<dyn SessionStore + Send + Sync>, session_store::Error> {
+        match self {
+            Self::Memory => Ok(Box::new(MemoryStore::new())),
+            #[cfg(feature = "json")]
+            Self::File { path } => Ok(Box::new(FileStore::new(path))),
+            #[cfg(feature = "cache")]
+            Self::Cache { ref uri } => match CacheType::from(uri.clone()) {
+                #[cfg(all(feature = "json", feature = "redis"))]
+                CacheType::Redis => Ok(Box::new(RedisStore::new(uri)?)),
+                CacheType::Unknown => Err(session_store::Error::Backend(format!(
+                    "Unsupported cache URI scheme: {}",
+                    uri.0.scheme()
+                ))),
+            },
+            #[cfg(feature = "db")]
+            Self::Database => {
+                unimplemented!();
```

Comment:
I ended up moving the DB implementation into another PR since there are some nuances to it; The DB option requires a migration model and generated migration, which I haven't gotten to the bottom of